src/lua/api-gateway/dogstatsd/Dogstatsd.lua

```diff
@@ -0,0 +1,100 @@
+
+--- Auxiliary module used to extract dogstats deamon initialization methods, incrementation of metrics
+-- and calls to other useful functions
+-- employed to measure calls to the Oauth provider: number, duration, etc.
+
+local Dogstatsd = {}
+
+function Dogstatsd:new(o)
+    o = o or {}
+    setmetatable(o, self)
+    self.__index = self
+    return o
+end
+
+--- Loads a lua gracefully. If the module doesn't exist the exception is caught, logged and the execution continues
+-- @param module path to the module to be loaded
+-- @return The loaded module, or nil if the module cannot be loaded
+--
+local function loadrequire(module)
+    ngx.log(ngx.DEBUG, "Loading module [" .. tostring(module) .. "]")
+    local function requiref(module)
+        return require(module)
+    end
+
+    local res, cls = pcall(requiref, module)
+    if not (res) then
+        ngx.log(ngx.WARN, "Could not load module [", module, "].")
+        return nil
+    end
+
+    return cls
+end
+
+local dogstatsd
+
+--- Returns an instance of dogstatsd only if it does not already exist. Returns the instance if the feature is enabled
+-- @param none
+-- @return An instance of dogstatsd or nil if the class cannot be instantiated
+--
+local function getDogstatsd()
+    if ngx.var.isDogstatsEnabled == nil or ngx.var.isDogstatsEnabled == "false" then
+        ngx.log(ngx.INFO, "dogstats module is disabled")
+        return nil
+    end
+
+    if dogstatsd ~= nil then
+        return dogstatsd
+    end
+
+    local restyDogstatsd = loadrequire('resty_dogstatsd')
+
+    if restyDogstatsd == nil then
+        ngx.log(ngx.WARN, "Could not loadrequire resty_dogstatsd.")
+        return nil
+    end
+
+    dogstatsd = restyDogstatsd.new({
+        statsd = {
+            host = "datadog.docker",
```

Comment:
The host should be able to be changed from Chef. The same thing applies to port. Make 8125 the default if the port is not specified in api-gateway-env-vars.server.conf